### ISSUE TITLE
LGTM and CodeCov setup

### ISSUE THREAD

**Issue by GlynLeine** [Rythe-Interactive/Legion-LLRI#81]
Using [https://lgtm.com/projects/g/Rythe-Interactive/Legion-LLRI/logs/languages/lang:cpp](https://lgtm.com/projects/g/Rythe-Interactive/Legion-LLRI/logs/languages/lang:cpp) you can setup code quality reports which might be neat.

also using [https://app.codecov.io/gh/Rythe-Interactive/Legion-LLRI](https://app.codecov.io/gh/Rythe-Interactive/Legion-LLRI) you can setup code coverage.

am planning to set it up for rythe as well once i'm done reworking the build process

**Comment by LeonBrands** [comment]
I think I'd like to get the analyse action from Rythe-Engine going in LLRI as well

**Comment by GlynLeine** [comment]
yeah i'm gonna make that work a bit better when i work on the build system. so then i can also look into getting that setup for LLRI

**Comment by LeonBrands** [comment]
LLRI just builds with cmake so that should be fairly straight-forward, if u need a reference for that, the current build action already does everything you need for project gen

**Comment by GlynLeine** [comment]
yeah i think i prolly just gotta bite the bullet and finally learn cmake...
